--- update-mime-database.c	2018-07-24 13:57:08.000000000 +0200
+++ update-mime-database.c	2019-06-22 19:47:43.953163738 +0200
@@ -18,6 +18,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <fcntl.h>
+#include <utime.h>
 
 #define XML_NS XML_XML_NAMESPACE
 #define XMLNS_NS "http://www.w3.org/2000/xmlns/"
@@ -94,6 +95,8 @@
 	 * with an unknown namespace.
 	 */
 	xmlDoc	*output;
+
+	time_t mtime;
 };
 
 struct _Glob {
@@ -262,6 +265,8 @@
 	xmlAddChild(root, xmlNewDocComment(type->output,
 		(xmlChar *)"Created automatically by update-mime-database. DO NOT EDIT!"));
 
+	type->mtime = 0;
+
 	for (i = 0; i < G_N_ELEMENTS(media_types); i++)
 	{
 		if (strcmp(media_types[i], type->media) == 0)
@@ -700,6 +705,8 @@
 {
 	xmlDoc *doc;
 	xmlNode *root, *node;
+	struct stat statbuf = {0};
+	time_t mtime = 0;
 
 	doc = xmlParseFile(filename);
 	if (!doc)
@@ -724,6 +731,10 @@
 		goto out;
 	}
 
+	if(stat(filename, &statbuf) != -1) {
+		mtime = statbuf.st_mtim.tv_sec;
+	}
+
 	for (node = root->xmlChildrenNode; node; node = node->next)
 	{
 		Type *type = NULL;
@@ -770,6 +781,8 @@
 				  filename, error->message);
 			g_error_free(error);
 		}
+
+		type->mtime = mtime;
 	}
 out:
 	xmlFreeDoc(doc);
@@ -1030,6 +1043,7 @@
 	char *media, *filename;
 	GError *local_error = NULL;
 	char *lower;
+	struct utimbuf times;
 
 	lower = g_ascii_strdown(type->media, -1);
 	media = g_strconcat(mime_dir, "/", lower, NULL);
@@ -1049,6 +1063,12 @@
 	if (!save_xml_file(type->output, filename, &local_error))
 		fatal_gerror(local_error);
 
+	if(type->mtime > 0) {
+		times.actime = type->mtime;
+		times.modtime = type->mtime;
+		utime(filename, &times);
+	}
+
 	if (!atomic_update(filename, &local_error))
 		fatal_gerror(local_error);
 
