diff --git a/squashfs-tools/xattr.c b/squashfs-tools/xattr.c
index b46550c..9e5845f 100644
--- a/squashfs-tools/xattr.c
+++ b/squashfs-tools/xattr.c
@@ -35,6 +35,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <sys/xattr.h>
+#include <linux/xattr.h>
 
 #include "squashfs_fs.h"
 #include "squashfs_swap.h"
@@ -206,6 +207,33 @@ static int read_xattrs_from_system(char *filename, struct xattr_list **xattrs)
 		}
 		xattr_list[i].vsize = vsize;
 
+		static const char XATTR_ESCAPE_PREFIX[] = "escape.";
+		if(
+			xattr_list[i].type == SQUASHFS_XATTR_USER
+			&& !strncmp(
+				xattr_list[i].full_name + XATTR_USER_PREFIX_LEN,
+				XATTR_ESCAPE_PREFIX,
+				sizeof(XATTR_ESCAPE_PREFIX)-1
+			)
+		) {
+			char *unesc = xattr_list[i].full_name;
+			xattr_list[i].type = get_prefix(
+				&xattr_list[i],
+				unesc
+				+ XATTR_USER_PREFIX_LEN
+				+ sizeof(XATTR_ESCAPE_PREFIX)-1
+			);
+			free(unesc);
+			// FIXME: is this correct?
+			if(xattr_list[i].type == -1) {
+				ERROR("Unrecognised xattr prefix %s\n",
+					xattr_list[i].full_name);
+				free(xattr_list[i].full_name);
+				i--;
+				continue;
+			}
+		}
+
 		TRACE("read_xattrs_from_system: filename %s, xattr name %s,"
 			" vsize %d\n", filename, xattr_list[i].full_name,
 			xattr_list[i].vsize);
