diff --git a/srcpkgs/xbps-triggers/files/gdk-pixbuf-loaders b/srcpkgs/xbps-triggers/files/gdk-pixbuf-loaders
index 26a7eee9..739bd4cd 100755
--- a/srcpkgs/xbps-triggers/files/gdk-pixbuf-loaders
+++ b/srcpkgs/xbps-triggers/files/gdk-pixbuf-loaders
@@ -36,7 +36,7 @@ run)
 		fi
 		if [ -x ${pixbuf_bin} ]; then
 			echo "Updating GDK Pixbuf loaders cache..."
-			${pixbuf_bin} --update-cache
+			../chroot ${pixbuf_bin} --update-cache
 		fi
 		;;
 	esac
diff --git a/srcpkgs/xbps-triggers/files/gtk-immodules b/srcpkgs/xbps-triggers/files/gtk-immodules
index 46e3a6bf..b4547ae6 100755
--- a/srcpkgs/xbps-triggers/files/gtk-immodules
+++ b/srcpkgs/xbps-triggers/files/gtk-immodules
@@ -38,7 +38,7 @@ run)
 		fi
 
 		echo "Updating GTK+ immodules cache..."
-		$immodules_bin --update-cache
+		../chroot $immodules_bin --update-cache
 		;;
 	esac
 	;;
diff --git a/srcpkgs/xbps-triggers/files/gtk3-immodules b/srcpkgs/xbps-triggers/files/gtk3-immodules
index 12042fc0..a4f3d7f8 100755
--- a/srcpkgs/xbps-triggers/files/gtk3-immodules
+++ b/srcpkgs/xbps-triggers/files/gtk3-immodules
@@ -38,7 +38,7 @@ run)
 		fi
 
 		echo "Updating GTK+3 immodules cache..."
-		$immodules_bin --update-cache
+		../chroot $immodules_bin --update-cache
 		;;
 	esac
 	;;
diff --git a/srcpkgs/xbps-triggers/files/info-files b/srcpkgs/xbps-triggers/files/info-files
index 7ad012b8..1575f79e 100755
--- a/srcpkgs/xbps-triggers/files/info-files
+++ b/srcpkgs/xbps-triggers/files/info-files
@@ -47,7 +47,7 @@ run)
 			;;
 		esac
 
-		$installinfo $infoargs ./$f $infodir/dir 2>/dev/null
+		../chroot $installinfo $infoargs ./$f $infodir/dir 2>/dev/null
 		if [ $? -eq 0 ]; then
 			echo "done."
 		else
diff --git a/srcpkgs/xbps-triggers/files/mimedb b/srcpkgs/xbps-triggers/files/mimedb
index e5535f24..d420d45e 100755
--- a/srcpkgs/xbps-triggers/files/mimedb
+++ b/srcpkgs/xbps-triggers/files/mimedb
@@ -28,7 +28,7 @@ run)
 		fi
 
 		echo "Updating shared-mime-info database..."
-		PKGSYSTEM_ENABLE_FSYNC=0 ${mimedb_bin} -n usr/share/mime > /dev/null
+		PKGSYSTEM_ENABLE_FSYNC=0 ../chroot ${mimedb_bin} -n usr/share/mime > /dev/null
 		;;
 	esac
 	;;
diff --git a/srcpkgs/xbps-triggers/files/system-accounts b/srcpkgs/xbps-triggers/files/system-accounts
index 77d840f2..39363bab 100755
--- a/srcpkgs/xbps-triggers/files/system-accounts
+++ b/srcpkgs/xbps-triggers/files/system-accounts
@@ -14,8 +14,6 @@ PKGNAME="$3"
 VERSION="$4"
 UPDATE="$5"
 
-export PATH="usr/sbin:usr/bin:/usr/sbin:/usr/bin:/sbin:/bin"
-
 group_add() {
 	local _grname _gid use_gid
 
@@ -26,11 +24,11 @@ group_add() {
 		use_gid="gid ${_gid}"
 	fi
 
-	if ! getent group ${_grname} >/dev/null; then
+	if ! ../chroot usr/bin/getent group ${_grname} >/dev/null; then
 		if [ -n "$use_gid" ]; then
-			groupadd -r ${_grname} -g ${_gid} >/dev/null 2>&1
+			unshare -r ../chroot usr/bin/groupadd -r ${_grname} -g ${_gid} >/dev/null 2>&1
 		else
-			groupadd -r ${_grname} >/dev/null 2>&1
+			unshare -r ../chroot usr/bin/groupadd -r ${_grname} >/dev/null 2>&1
 		fi
 		if [ $? -eq 0 ]; then
 			echo "Created ${_grname} ($use_gid) system group."
@@ -50,31 +48,10 @@ run)
 		exit 0
 	fi
 
-	if [ -x sbin/useradd -o -x bin/useradd ]; then
-		USERADD=1
-	fi
-	if [ -x sbin/usermod -o -x bin/usermod ]; then
-		USERMOD=1
-	fi
-	if [ -x sbin/groupadd -o -x bin/groupadd ]; then
-		GROUPADD=1
-	fi
-	if [ -x bin/getent -o -x sbin/getent ]; then
-		GETENT=1
-	fi
-	if [ -x bin/passwd -o -x sbin/passwd ]; then
-		PASSWD=1
-	fi
-
 	case "$TARGET" in
 	post-install)
 		# System groups required by a package.
 		for grp in ${system_groups}; do
-			if [ -z "$GROUPADD" -a -z "$GETENT" ]; then
-				echo "WARNING: cannot create ${grp} system group (missing groupadd/getent)"
-				echo "The following group must be created manually: $grp"
-				continue
-			fi
 			group_add $grp
 		done
 
@@ -94,23 +71,12 @@ run)
 			[ -z "$descr" ] && descr="${_uname} unprivileged user"
 			[ -n "$groups" ] && user_groups="-G $groups"
 
-			if [ -z "$USERADD" -a -z "$GETENT" -a -z "$PASSWD" ]; then
-				echo "WARNING: cannot create ${acct} system user/group (missing useradd/getent/passwd)"
-				echo "The following system account must be created:"
-				echo "   Account: ${uname:-${_uid}} (uid: '${_uid}')"
-				echo "   Description: '${descr}'"
-				echo "   Homedir: '${homedir}'"
-				echo "   Shell: '${shell}'"
-				echo "   Additional groups: '${groups}'"
-				continue
-			fi
-
 			group_add ${pgroup:-${acct}}
 
-			if ! getent passwd ${_uname} >/dev/null; then
-				useradd -c "$descr" -d "$homedir" -s "$shell" ${user_groups} \
+			if ! ../chroot usr/bin/getent passwd ${_uname} >/dev/null; then
+				unshare -r ../chroot usr/bin/useradd -c "$descr" -d "$homedir" -s "$shell" ${user_groups} \
 					${pgroup:+-N} ${use_id:=-g ${pgroup:-${_uname}}} -r ${_uname} && \
-					passwd -l ${_uname} >/dev/null 2>&1
+					unshare -r ../chroot usr/bin/passwd -l ${_uname} >/dev/null 2>&1
 				if [ $? -eq 0 ]; then
 					echo "Created ${_uname} (${_uid}) system user."
 				else
diff --git a/srcpkgs/xbps-triggers/files/update-desktopdb b/srcpkgs/xbps-triggers/files/update-desktopdb
index bc447ec8..629cde5a 100755
--- a/srcpkgs/xbps-triggers/files/update-desktopdb
+++ b/srcpkgs/xbps-triggers/files/update-desktopdb
@@ -36,7 +36,7 @@ run)
 
 		if [ -d $desktopdb_dir ]; then
 			echo "Updating MIME database..."
-			${desktopdb_bin} ${desktopdb_dir}
+			../chroot ${desktopdb_bin} ${desktopdb_dir}
 		fi
 		;;
 	esac
