--- a/srcpkgs/xbps-triggers/files/pycompile
+++ b/srcpkgs/xbps-triggers/files/pycompile
@@ -64,7 +64,31 @@ compile()
 				usr/lib/python${pycompile_version}/site-packages/${f}
 		fi
 	done
-	update_ldcache
+}
+
+redate()
+{
+	local timestamp_to_mtime='touch -c -d "@$(hexdump -s4 -n4 -e \"%d\" "$1")" "$1"'
+	local parent_mtime_to_mtime="touch -c -r {}/.. {}"
+
+	for f in ${pycompile_dirs}; do
+		echo "Changing mtime of byte-compiled python${pycompile_version} files in ${f}..."
+		find ${f} -type f -name \*.py[co] -exec sh -c "$timestamp_to_mtime" -- '{}' \; 2>&1 >/dev/null
+		find ${f} -type d -name __pycache__ -exec $parent_mtime_to_mtime \; 2>&1 >/dev/null
+	done
+	for f in ${pycompile_module}; do
+		echo "Changing mtime of byte-compiled python${pycompile_version} code for module ${f}..."
+		if [ -d usr/lib/python${pycompile_version}/site-packages/${f} ]; then
+			find usr/lib/python${pycompile_version}/site-packages/${f} \
+				-type f -name \*.py[co] -exec sh -c "$timestamp_to_mtime" -- '{}' \; 2>&1 >/dev/null
+			find usr/lib/python${pycompile_version}/site-packages/${f} \
+				-type d -name __pycache__ -exec $parent_mtime_to_mtime \; 2>&1 >/dev/null
+		else
+			for i in usr/lib/python${pycompile_version}/site-packages/${f%.py}.py[co]; do
+				sh -c "$timestamp_to_mtime" -- "$i"
+			done
+		fi
+	done
 }
 
 remove()
@@ -85,7 +109,6 @@ remove()
 			rm -f usr/lib/python${pycompile_version}/site-packages/${f%.py}.py[co]
 		fi
 	done
-	update_ldcache
 }
 
 case "$ACTION" in
@@ -99,9 +122,12 @@ run)
 	case "$TARGET" in
 	post-install)
 		compile
+		redate
+		update_ldcache
 		;;
 	pre-remove)
 		remove
+		update_ldcache
 		;;
 	*)
 		exit 1
